docs/source/deploying-volttron/secure-deployment-considerations.rst

```diff
@@ -0,0 +1,89 @@
+
+Security of computing systems is a complex topic which depends not only on the
+security of each component, but on how software components interact and on the
+environment in which they are running.
+In the subsections here, we will discuss a variety of possible actions which
+may increase the security of a particular deployment, along with their context.
+
+For more examples and discussion, see the `Publications section of the VOLTTRON website <https://volttron.org/publications>`_
+where there are a number of Threat Profile reports.
+
+Running as a Managed System Process
+===================================
+
+It is possible that the running VOLTTRON process could exit undesirably (either due
+to a bug, or some malicious action).
+For scenarios where not having the VOLTTRON process running presents a business
+risk, it is recommended to follow the :ref:`system service setup`
+to leverage the host system's process monitoring and management system.
+Under this configuration, the system will be configured to restart VOLTTRON in the
+event that it fails.
+
+.. note::
+
+    For this configuration to be effective, it is important that the platfrom
+    is configured such that it automatically starts up in the desired state.
+    In particular, review the installed agents and be sure that agents which
+    should be running are "enabled" and that their priorities are set such
+    that they start in the intended order.
+
+There are scenarios when this configuration may not be desired:
+
+1. If a system restarts cleanly after an unexpected failure, it is possible that
+   the underlying issue could go unnoticed (and therefore unresolved). This would
+   happen if a user checks the system and sees it is running but does not have a
+   way to realize that there has been one or more restarts. For development systems
+   it may be desirable to not restart, leaving the system in a failed state which
+   is more likely to be noticed as unusual, and with the failure details still present
+   in the recent logs. Consider the relative value of platform up-time and failure
+   this kind of failure discovery. If both are highly valuable, it may be possible
+   to add extra notifications to the process monitoring system (systemd, initd, or
+   other) so that records are retained while service is restored.
+2. For development systems, or systems that are frequently stopped or restarted,
+   it can be more convenient to use the normal start and stop scripts packaged
+   with VOLTTRON. These do not require the user have system-level permissions
+   and are easily used from the terminal.
+
+
+Run Web Server Behind Proxy
+===========================
+
+A VOLTTRON deployment may be web-enabled, allowing various interactions over HTTP.
+There are many reasons why it is often desirable to deploy an external reverse
+proxy in front of the system, including:
+
+- Allows regular security patching of the exposed web server independent of the VOLTTRON
```

Comment:
Load balancing is another reason...though its a fan-in pattern rather than fan out.

Reply:
expanded